### ISSUE TITLE
bizarre comparison bug

### ISSUE THREAD

**Issue by pzinn** [Macaulay2/M2#792]
Can anyone else confirm the following bug I have (on my own compiled version):
```
i1 : -1. ? 0

o1 = <

o1 : Keyword

i2 : {0./0.}

o2 = {NotANumber}

o2 : List

i3 : -1. ? 0

o3 = incomparable

o3 : Symbol
```

**Comment by DanGrayson** [comment]
Yes, that's strange.  Here's simpler way to evoke it, based on the observation that the hash code
of a list is precomputed and stored in the list, so putting something in a list causes its hash code to be computed:
```
i1 : -1. ? 0

o1 = <

o1 : Keyword

i2 : 0./0.

o2 = NotANumber

o2 : RR (of precision 53)

i3 : -1. ? 0

o3 = <

o3 : Keyword

i4 : hash ooo

o4 = 843

i5 : -1. ? 0

o5 = incomparable

o5 : Symbol
```
The code for "hash" applied to a real number is mine and is this:
```
int mpfr_hash(mpfr_t x) {
  int h = 0;
  int n = (x->_mpfr_prec+mp_bits_per_limb-1)/mp_bits_per_limb;
  int i;
  if (0 != mpfr_sgn(x))
    for (i = 0; i<n; i++, h*=3737) h += x->_mpfr_d[i];
  return 777 + h * 3737 + x->_mpfr_exp + 11 * x->_mpfr_sign;
}
```
It has no side effects!

**Comment by pzinn** [comment]
the culprit seems to be L223 (or L228, L274, L279, L288, L293, L298, L303, etc) of actors3.d:
```
	       if flagged() then incomparableE else
```
the flag must be somehow triggered by the hash of 0./0. ?

**Comment by DanGrayson** [comment]
Thanks!  I've fixed it in commit b473f2f0a271cfd155961e514eeaf553f4d306f6
